osu.Game/Database/EFToRealmMigrator.cs

```diff
@@ -35,6 +35,16 @@ public void Run()
 
         private void migrateSkins(DatabaseWriteUsage db)
         {
+            // can be removed 20220530.
+            var existingSkins = db.Context.SkinInfo
+                                  .Include(s => s.Files)
+                                  .ThenInclude(f => f.FileInfo)
+                                  .ToList();
+
+            // previous entries in EF are removed post migration.
+            if (!existingSkins.Any())
+                return;
+
             var userSkinChoice = config.GetBindable<string>(OsuSetting.Skin);
             int.TryParse(userSkinChoice.Value, out int userSkinInt);
```

Comment:
Ugh, I wrote this with the intention of the `switch` only being run if this succeeded. That way it would be safe regardless of migration state.